### PR TITLE
Added legend functionality to Horizontal Bar chart

### PR DESCRIPTION
**Pull request by Anush2303**
## Previous Behavior

No legends in chart.

## New Behavior

Legends functional in chart now.

## Related Issue(s)

- Fixes #

### CODE REVIEW

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts [hunk 1]
```diff
@@ -75,13 +77,54 @@ export const styles: ElementStyles = css`
...
     justify-content: space-between;
     text-overflow: ellipsis;
     overflow: hidden;
     white-space: nowrap;
     display: block;
-    color: theme.palette.neutralPrimary;
+    font-size: 14px;
+    font-weight: 400px;
+    line-height: 16px;
+    color: rgb(50, 49, 48);
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:93]: instead of hardcoding use the design tokens for here. https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/theme/design-tokens.ts

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts [hunk 2]
```diff
@@ -75,13 +77,54 @@ export const styles: ElementStyles = css`
...
+    width: 12px;
+    height: 12px;
+    margin-right: 8px;
+    border: 1px solid;
+  }
+  .legendText {
+    font-family: ${fontFamilyBase};
+    font-size: ${fontSizeBase200};
+    font-weight: ${fontWeightRegular};
+    line-height: ${lineHeightBase200};
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:124]: replace these group with typography defined in this file. https://github.com/microsoft/fluentui/blob/master/packages/web-components/src/styles/partials/typography.partials.ts

For eg all the font related styles here can be replaced with this
https://github.com/microsoft/fluentui/blob/263ead68a05ac23499be137b41e02a5579522c08/packages/web-components/src/styles/partials/typography.partials.ts#L56

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts [hunk 3]
```diff
@@ -115,6 +118,26 @@ export class HorizontalBarChart extends FASTElement {
       },
     ];
 
+    const allchartPoints = [...chartPoints1, ...chartPoints2, ...chartPoints3];
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:121]: these should be part of the story not the .ts file

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts (hunk 1 above)
Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:93]: Okay

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts [hunk 4]
```diff
@@ -220,6 +243,61 @@ export class HorizontalBarChart extends FASTElement {
...
+      buttons[i].addEventListener('mouseover', () => {
+        for (let j = 0; j < bars!.length; j++) {
+          if (bars![j].getAttribute('barinfo') !== buttons[i].textContent) {
+            bars![j].style['opacity'] = '0.1';
+          }
+        }
+        for (let j = 0; j < buttons.length; j++) {
+          if (j !== i) {
+            const legendRect = buttons[j].getElementsByClassName('legendRect')[0];
+            legendRect.style['backgroundColor'] = 'transparent';
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:281]: instead of setting the style manually, could you put the style in a class and assign the correct class or subclass

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts (hunk 2 above)
Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:124]: Okay got it. But this works as a group. In case we need different styles within one group?

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts [hunk 5]
```diff
@@ -220,6 +243,61 @@ export class HorizontalBarChart extends FASTElement {
...
+
+            const legendText = buttons[j].getElementsByClassName('legendText')[0];
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:283]: use ref instead of get elements by classname

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts (hunk 4 above)
Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:281]: Okay

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts [hunk 6]
```diff
@@ -220,6 +243,61 @@ export class HorizontalBarChart extends FASTElement {
...
+            legendText.style['opacity'] = '0.67';
+          }
+        }
+      });
+      buttons[i].addEventListener('mouseout', () => {
+        for (let j = 0; j < bars!.length; j++) {
+          bars![j].style['opacity'] = '1';
+        }
+        for (let j = 0; j < buttons.length; j++) {
+          const legendRect = buttons[j].getElementsByClassName('legendRect')[0];
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:293]: add error handling if the element being queried are not found

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.ts (hunk 3 above)
Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.ts:121]: Yes for now, we have defined data also in .ts file only. Moving forward when will shift that to the stories file, then we can move this as well.

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts (hunk 2 above)
Comment (AtishayMsft) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:124]: we can override any specific style element if needed. Though it would be a corner case. We want to follow the standard design tokens everywhere.

Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.styles.ts:124]: Okay got it.

**Review by AtishayMsft — APPROVED** [review]

File: packages/web-components/src/horizontalbarchart/horizontalbarchart.stories.ts [hunk 7]
```diff
@@ -13,6 +13,12 @@ export default {
 
 export const Basic: Story<FluentHorizontalBarChart> = renderComponent(storyTemplate).bind({});
 
+export const NMVariant: Story<FluentHorizontalBarChart> = renderComponent(html<StoryArgs<FluentHorizontalBarChart>>`
```
Comment (AtishayMsft) [inline packages/web-components/src/horizontalbarchart/horizontalbarchart.stories.ts:16]: lets move NM variant to a separate PR and merge this one.

Comment (Anush2303) [reply packages/web-components/src/horizontalbarchart/horizontalbarchart.stories.ts:16]: Okay sure
